### PR TITLE
[CDAP-15097] artifact version defaults to 1.0.0-SNAPSHOT, if not present

### PR DESCRIPTION
**Pull request by itsanudeep**
When uploading an artifact in wrangler,

- If name is present and version is present i.e abc-1.2.3.jar -> no change; name: abc, version: 1.2.3
- If name is present and version is not i.e abc.jar -> name: abc, version: 1.0.0-SNAPSHOT
- If name is not present and version is present i.e 1.2.3.jar -> name: 1.2.3, version: 1.2.3

Also made changes to use helper method to get artifact's name and version.

JIRA: https://issues.cask.co/browse/CDAP-15097

### CODE REVIEW

**Comment by itsanudeep** [comment]
BUILD: https://builds.cask.co/browse/CDAP-UDUT335

**Comment by chtyim** [comment]
The backend actually default to "-SNAPSHOT" where there is no version (https://github.com/cdapio/cdap/blob/develop/cdap-proto/src/main/java/io/cdap/cdap/proto/id/ApplicationId.java#L34)

It will be good to do the same.

**Comment by ajainarayanan** [comment]
@chtyim Right now UI parses the file name to pass appropriate artifact information in headers. Is it possible to not have the UI do this and have this logic in backend if only name is passed? Thinking if it would help avoid duplicating this logic both in FE and BE

**Comment by chtyim** [comment]
Yes, we should do it in the backend if it not yet done. @albertshau ?

**Comment by albertshau** [comment]
Is this for the POST /namespaces/{namespace-id}/artifacts/{artifact-name} endpoint?

I suppose we could make it POST /namespaces/{namespace-id}/artifacts and pass the filename as a header. But it would mean we also need to change the reload system artifacts endpoint from POST /namespaces/system/artifacts to POST /artifacts or something like that.

But looking at the jira that prompted this... I don't think it's right to default to a snapshot version if none is present in the filename. Which JDBC driver was causing this problem? Version is an important piece of information. If it defaults to a snapshot version, somebody can upload two completely different jars and the latest one will clobber the earlier one.

**Comment by elfenheart** [comment]
@albertshau It was the JDBC driver for SAP HANA that prompted this JIRA: `ngdbc.jar`

**Comment by albertshau** [comment]
Is there more than one version of that driver? It seems like we need to modify the user experience for these cases. It might not be that the add artifact endpoint needs to take a filename, but that the UI needs a way to ask the backend how to parse a filename and behave a little differently if there is no version.

**Comment by itsanudeep** [comment]
@elfenheart Users can now specify driver/library version during upload. The field defaults to '1.0.0'.

File: cdap-ui/app/cdap/services/__tests__/helpers.test.js [hunk 1]
```diff
@@ -57,4 +57,10 @@ describe('Unit Tests for Helpers: "getArtifactNameAndVersion"', () => {
     expect(version).toBe(undefined);
   });
 
+  it('Usecase 7: Should return filename for name and version when filename is the version i.e 1.2.3.jar', () => {
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/services/__tests__/helpers.test.js:60]: 👏 👏 On making sure to add a unit test and fixing existing one based on the change. Kudos @itsanudeep

File: cdap-ui/app/cdap/services/helpers.js [hunk 2]
```diff
@@ -215,11 +215,23 @@ function getArtifactNameAndVersion(nameWithVersion) {
...
+      return { name: nameWithVersion, version: nameWithVersion };
+    }
+  }
+  let name = version
+    ? nameWithVersion.substr(0, nameWithVersion.indexOf(version) - 1)
+    : nameWithVersion;
+  // if version is not present, default it to 1.0.0-SNAPSHOT
   if (!version) {
-    return { name: nameWithVersion, version: undefined };
+    version = '1.0.0-SNAPSHOT';
```
Comment (elfenheart) [inline cdap-ui/app/cdap/services/helpers.js]: this should be 1.0.0, without the SNAPSHOT

Comment (itsanudeep) [reply cdap-ui/app/cdap/services/helpers.js]: fixed.

**Review by elfenheart — APPROVED** [review]
👍 LGTM

**Pull request merged**